test/integration/TokenAssociateIntegrationTest.js

```diff
@@ -134,6 +141,752 @@ describe("TokenAssociate", function () {
+                ).tokens.get(tokenId);
+
+                const newTokenBalance2 = (
+                    await new AccountBalanceQuery()
+                        .setAccountId(receiverId)
+                        .execute(env.client)
+                ).tokens.get(tokenId2);
+
+                expect(newTokenBalance.toInt()).to.equal(1);
+                expect(newTokenBalance2.toInt()).to.equal(1);
+            });
+
+            it("receiver should have token balance even if it has given allowance to spender", async function () {
+                this.timeout(120000);
+                const spenderKey = PrivateKey.generateECDSA();
+
+                const unlimitedAutoAssociationTx =
+                    await new AccountUpdateTransaction()
+                        .setAccountId(receiverId)
+                        .setMaxAutomaticTokenAssociations(-1)
+                        .freezeWith(env.client)
+                        .sign(receiverKey);
+
+                await (
+                    await unlimitedAutoAssociationTx.execute(env.client)
+                ).getReceipt(env.client);
+
+                const spenderAccountCreateTx =
+                    await new AccountCreateTransaction()
+                        .setKey(spenderKey)
+                        .setInitialBalance(new Hbar(1))
+                        .execute(env.client);
+
+                const spenderId = (
+                    await spenderAccountCreateTx.getReceipt(env.client)
+                ).accountId;
+
+                const tokenCreateResponse = await new TokenCreateTransaction()
+                    .setTokenName("ffff")
+                    .setTokenSymbol("F")
+                    .setInitialSupply(TOKEN_SUPPLY)
+                    .setTreasuryAccountId(env.operatorId)
+                    .setAdminKey(env.operatorKey)
+                    .setSupplyKey(env.operatorKey)
+                    .execute(env.client);
+
+                const { tokenId } = await tokenCreateResponse.getReceipt(
+                    env.client,
+                );
+
+                const tokenTransferHalfSupply = await new TransferTransaction()
+                    .addTokenTransfer(tokenId, env.operatorId, -TRANSFER_AMOUNT)
+                    .addTokenTransfer(tokenId, receiverId, TRANSFER_AMOUNT)
+                    .execute(env.client);
+
+                await tokenTransferHalfSupply.getReceipt(env.client);
+
+                const tokenAllowanceTx =
+                    await new AccountAllowanceApproveTransaction()
+                        .approveTokenAllowance(
+                            tokenId,
+                            receiverId,
+                            spenderId,
+                            1,
+                        )
+                        .freezeWith(env.client)
+                        .sign(receiverKey);
+
+                await (
+                    await tokenAllowanceTx.execute(env.client)
+                ).getReceipt(env.client);
```

Comment:
The approval should be before the transfer. Check out my latest commit [here](https://github.com/hashgraph/hedera-sdk-go/pull/980/commits/f71d392b9b6fa9b9fe8fb9ffbfc32b2930e1ff8d), since it's a bit tricky to pull this off

Reply:
A bit tricky is the least we can say 😆 At least it was easier than NFT approval transfer.